### ISSUE TITLE
using pip --no-dependencies to fix mac-pro/anaconda install bug

### ISSUE THREAD

**Issue by rcherukuri12** [deepmind/sonnet#20]
Machine : Mac Pro
Tensorflow 1.0.1 installed with method suggested for Anaconda ( in TensorFlow docs ).
Build and install Sonnet you get the following error even if correct TF is present:
pip install /tmp/sonnet/*.whl
Processing /tmp/sonnet/sonnet-1.0-py3-none-any.whl
Collecting nose-parameterized>=0.6.0 (from sonnet==1.0)
  Downloading nose_parameterized-0.6.0-py2.py3-none-any.whl
Collecting tensorflow>=1.0.1 (from sonnet==1.0)
  Could not find a version that satisfies the requirement tensorflow>=1.0.1 (from sonnet==1.0) (from versions: 0.12.1, 1.0.0, 1.1.0rc0, 1.1.0rc1, 1.1.0rc2)
No matching distribution found for tensorflow>=1.0.1 (from sonnet==1.0)

work-around:
pip install /tmp/sonnet/*.whl --no-dependencies

**Comment by hannes-brt** [comment]
I have also noticed that the `tensorflow` requirement in `install_requires=` causes some issues. In particular, I was using Sonnet on Google Cloud ML Engine. Their VM already has an optimized and GPU-enabled TensorFlow version installed. But installing the Sonnet wheel caused the machine to download the CPU-only `tensorflow` package from PyPI and shadowing the existing TensorFlow installation, resulting in the GPU not being used. I had to remove the `tensorflow` line from `setup.py` to make it work.
